Purge Facebook

phillipthelen opened the pull request:
This is the second PR matching with #13683.

This actually removes most code references to Facebook and removes it as an option for social auth. It does not delete the database structure, since we probably want to keep that until we are sure everything works correctly and no issues would arise from deleting that data.

This should not be merged in/deployed yet. This is for when we are ready to do this step.

phillipthelen committed:
Don't sign in user when trying to connect a social account that was already created

phillipthelen committed:
Log social users into matching local auth accounts

If the social account has an email that already exists as a local user, instead of creating a new account log them into their account and add the social auth to the account

phillipthelen committed:
If possible set local authentication email for social users

phillipthelen committed:
Allow password reset emails to be sent to social login users

phillipthelen committed:
lint fixes

phillipthelen committed:
Fix issues and tests

phillipthelen committed:
fix tests

phillipthelen committed:
Fix lint error.

phillipthelen committed:
Fix lint error.

phillipthelen committed:
purge Facebook.

Only keep it in some select places to allow for some compatablilty.

phillipthelen committed:
purge Facebook.

Only keep it in some select places to allow for some compatablilty.

phillipthelen committed:
Fix error

phillipthelen committed:
Fix error

phillipthelen committed:
Merge branch 'develop' into exbook2_electric_boogaloo

phillipthelen committed:
fix error

phillipthelen committed:
fix error

phillipthelen commented:
The relevant tests all pass. The error is coming from a test that is failing due to g1g1 right now.

phillipthelen committed:
Let settings handle it when you don't have a password set but an email

phillipthelen committed:
fix error

phillipthelen committed:
Fix boolean logic

phillipthelen committed:
fix json conversion

phillipthelen committed:
.

phillipthelen committed:
fix password reset for old social accounts

SabreCat commented on website/client/src/components/appFooter.vue:
```diff
@@ -84,8 +84,8 @@
               <a
-                @click.prevent="openBugReportModal()"
                 target="_blank"
+                @click.prevent="openBugReportModal()"
```
This probably doesn't need to be in this PR, especially since we're rewriting the footer

SabreCat commented on website/client/src/components/bugReportModal.vue:
```diff
@@ -12,12 +12,18 @@
         {{ $t('reportBug') }}
       </h2>
 
-      <div v-once class="report-bug-header-describe">
```
This file is likewise unrelated (probably dragged in via a linter autofix?)

SabreCat commented on website/client/src/components/bugReportSuccessModal.vue:
```diff
@@ -17,15 +17,22 @@
     </div>
     <div>
-      <span class="svg-icon check-icon"
-            v-html="icons.checkCircleIcon"
```
extraneous file

SabreCat commented on website/client/src/components/groups/tavern.vue:
```diff
@@ -361,8 +361,8 @@
               <a
-                @click.prevent="openBugReportModal()"
                 target="_blank"
+                @click.prevent="openBugReportModal()"
```
extraneous file

SabreCat commented on website/client/src/components/header/menu.vue:
```diff
@@ -311,8 +311,8 @@
               </router-link>
               <a
                 class="topbar-dropdown-item dropdown-item"
-                @click.prevent="openBugReportModal()"
```
'nother 😅

SabreCat commented on website/client/src/components/header/notifications/worldBoss.vue:
```diff
@@ -178,12 +178,12 @@ import sword from '@/assets/svg/sword.svg';
 import { worldStateMixin } from '@/mixins/worldState';
 
 export default {
-  mixins: [
```
'n' another

SabreCat commented on website/client/src/components/inventory/items/index.vue:
```diff
@@ -194,7 +194,10 @@
                     {{ context.item.text }}
                   </h4>
-                  <questInfo :quest="context.item" :purchased="true" />
+                  <questInfo
```
blap

SabreCat commented on website/client/src/components/shared/closeIcon.vue:
```diff
@@ -1,12 +1,12 @@
 <template>
   <button
     title="close dialog"
-    @click="$emit('click', $event)"
```
lint why

SabreCat commented on website/client/src/components/snackbars/notification.vue:
```diff
@@ -1,10 +1,17 @@
 <template>
   <div class="notification-animation-holder">
-    <div class="notification-holder"
```
...

SabreCat commented on website/client/src/components/snackbars/notifications.vue:
```diff
@@ -81,12 +81,12 @@ const DELAY_DELETE_AND_NEW = 60;
 const DELAY_FILLING_ENTRIES = 240;
 
 export default {
-  mixins: [
```
😭

SabreCat commented on website/client/src/components/static/contact.vue:
```diff
@@ -10,8 +10,8 @@
             <a
-              @click.prevent="openBugReportModal()"
               target="_blank"
+              @click.prevent="openBugReportModal()"
```
🙄

SabreCat commented on website/client/src/components/static/privacy.vue:
```diff
@@ -8,7 +8,7 @@
     <p>
       HabitRPG, Inc. (“HabitRPG,” “we,” “us,” or “our”) welcomes you. This privacy notice (the “Privacy
       Notice”) describes how we process the information we collect about or from you through our Website
-      located at <a href='https://habitica.com/static/home'>https://habitica.com/static/home</a> and/or our Apps
```
I'm scanning through this for anything Facebook related but not seeing it... looks like another aggressive lint

SabreCat commented on website/client/src/components/static/terms.vue:
```diff
@@ -2,11 +2,20 @@
   <!-- eslint-disable max-len -->
   <div class="container-fluid">
     <h1>Terms of Service</h1>
-    <p class="strong pagemeta">Last Updated: December 14, 2021</p>
```
same here I think?

SabreCat reviewed (changes requested):
I think the intended changes here are good, but there's a bunch of unrelated stuff that ought to be in a separate PR. Steps to me look like

* Remove the files that have nothing to do with Facebook--looks like mostly client side code style adjustments
* Merge develop back and resolve conflicts
* If desired, submit a separate lint-fix PR

I'm guessing that will clean up everything necessary! Some of the changes here look like things we did in the prereq social auth improvement PRs that maybe didn't get split out? But the develop merge should straighten that out, though if there are discrepancies we'll need to dig into and address those!

phillipthelen committed:
Revert "lint fixes"

This reverts commit c244b1651c2c23050d058fe15381518231d8a317.

# Conflicts:
#	website/client/src/components/auth/registerLoginReset.vue
#	website/client/src/components/static/contact.vue

phillipthelen committed:
Don't sign in user when trying to connect a social account that was already created

phillipthelen committed:
Log social users into matching local auth accounts

If the social account has an email that already exists as a local user, instead of creating a new account log them into their account and add the social auth to the account

phillipthelen committed:
If possible set local authentication email for social users

phillipthelen committed:
Allow password reset emails to be sent to social login users

phillipthelen committed:
lint fixes

phillipthelen committed:
Fix issues and tests

phillipthelen committed:
fix tests

phillipthelen committed:
Let settings handle it when you don't have a password set but an email

phillipthelen committed:
fix error

phillipthelen committed:
Fix boolean logic

phillipthelen committed:
fix json conversion

phillipthelen committed:
fix password reset for old social accounts

phillipthelen committed:
Revert "fix password reset for old social accounts"

This reverts commit 7e0069a80fd20bc54d13cec4f83e072ad20d6610.

phillipthelen committed:
Merge branch 'exbook2_electric_boogaloo' of https://github.com/HabitRPG/habitica into exbook2_electric_boogaloo

# Conflicts:
#	website/client/src/components/static/privacy.vue
#	website/client/src/components/static/terms.vue
#	website/common/locales/en/settings.json
#	website/server/controllers/api-v3/auth.js
#	website/server/libs/auth/index.js
#	website/server/libs/auth/social.js

phillipthelen committed:
fix duplicate code

SabreCat committed:
Merge branch 'develop' into exbook2_electric_boogaloo

SabreCat committed:
chore(misc): remove irrelevant changes

SabreCat committed:
chore(privacy): update policy page with note about FB
